.github/workflows/build.yaml

```diff
@@ -15,41 +15,66 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
-      - name: Set up Python
-        uses: actions/setup-python@v2
+      - uses: conda-incubator/setup-miniconda@v2
         with:
           python-version: 3.8
-
-      - name: Install sqlite
-        run: sudo apt-get install sqlite libyaml-dev
+          channels: conda-forge,defaults
+          channel-priority: strict
+          show-channel-urls: true
 
       - name: Update pip/wheel infrastructure
         run: |
-          python -m pip install --upgrade pip
-          pip install wheel
+          conda install -y -q pip wheel
+
+      - name: Install sqlite
+        shell: bash -l {0}
+        run: |
+          conda install -y -q sqlite
 
       - name: Install postgresql (server)
         run: sudo apt install postgresql
 
       - name: Install postgresql Python packages
-        run: pip install psycopg2 testing.postgresql
+        shell: bash -l {0}
+        run: |
+          conda install -y -q psycopg2
+          pip install testing.postgresql
 
       - name: Install cryptography package for moto
-        run: pip install cryptography
+        shell: bash -l {0}
+        run: |
+          conda install -y -q cryptography
 
       - name: Install WebDAV packages for testing
-        run: pip install cheroot wsgidav
+        shell: bash -l {0}
+        run: |
+          pip install cheroot wsgidav
 
       - name: Install dependencies
+        shell: bash -l {0}
         run: |
           pip install -r requirements.txt
```

Comment:
I would recommend `conda install --quiet -y --file=requirements.txt` here if we want to move to an all-conda testbed rather than getting conda and pip mixed.  But at the same time I can see the argument for only using conda to do the non-pypi packages to replace the old ubuntu versions.

Reply:
I tried `conda` but it does not know how to handle `git+git://github.com/lsst/sphgeom@master#egg=lsst_sphgeom` which is in our `requirements.txt`

Reply:
Oh good point.  We'd have to do some more rearchitecting to make that work, and is definitely not worth it at this point.  Something like https://stackoverflow.com/questions/59864503/direct-link-to-github-in-requirements-txt-while-using-conda